### ISSUE TITLE
Layer Visible Issues

### ISSUE THREAD

**Issue by kpetrow** [gdsestimating/dxf-parser#21]
Awesome library.  I wrote my own parser not knowing this was out there and month later decided to abandon ship for this great library.  The final straw was code 38 elevations for polylines instead of code 30 for vertex elevations....  THANKS!

for reference i use the dxf manual found [here](http://images.autodesk.com/adsk/files/autocad_2012_pdf_dxf-reference_enu.pdf)

In the following example one layer, is invisible(negative code 62 color), one layer is frozen and should not be visible( value 1 for code 70) and one layer is visible(positive value for code 62 and value zero for code 70)
DXF snippet:

> 100
> AcDbSymbolTableRecord
> 100
> AcDbLayerTableRecord
>   2
> EXISTING
>  70
>      0 **_// not frozen and visible_**
>  62
>      2 **_//positive and visible_**
>   6
> DASHED
> 370
>     -3
> 390
> F
>   0
> LAYER
>   5
> 3C
> 102
> {ACAD_XDICTIONARY
> 360
> 42A
> 102
> }
> 330
> 2
> 100
> AcDbSymbolTableRecord
> 100
> AcDbLayerTableRecord
>   2
> FINAL
>  70
>      0 **_// not frozen and visible_**
>  62
>     -1  _**// negative color means layer FINAL off**_ 
>   6
> Continuous
> 370
>     -3
> 390
> F
>   0
> LAYER
>   5
> 3D
> 102
> {ACAD_XDICTIONARY
> 360
> 42C
> 102
> }
> 330
> 2
> 100
> AcDbSymbolTableRecord
> 100
> AcDbLayerTableRecord
>   2
> PNTDESC
>  70
>      1 **_// layer PNTDESC should be frozen and invisible_**
>  62
>      3
>   6
> Continuous
> 370
>     -3
> 390
> F
>   0
> LAYER
>   5
> 3F
> 102
> {ACAD_XDICTIONARY
> 360
> 430
> 102
> }
> 330
> 2

JSON snippet:

> "layer": {
> 			"handle": "2",
> 			"ownerHandle": "0",
> 			"layers": {
> 				"EXISTING": {
> 					"name": "EXISTING",
> 					"visible": false, //This is the one visible layer (code 70 = 0 && code 62 is positive
> 					"color": 16776960
> 				},
> 				"FINAL": {
> 					"name": "FINAL",
> 					"visible": true, //  this layer has negative color means not visible
> 					"color": 16711680
> 				},
>                             "PNTDESC": {
> 					"name": "PNTDESC",
> 					"visible": false, // this layer is frozen should not be visible
> 					"color": 65280
> 				},

looks like line 533 of DxfParser.js is backwards:
`layer.visible = curr.value <= 0;`
should be 
`layer.visible = curr.value > 0;`

And you do not check for frozen layers??

possibly add in this code into parseLayers.  
```
case 70: // frozen layer
     layer.frozen = (curr.value != 1 &&  curr.value != 2 );
     curr = scanner.next();
     break;
```
OR dirtier:
```
case 62: // color, visibility
     layer.visible = (layer.visible && curr.value > 0);` // code 70 is always prior to 62
     layer.color = getAcadColor(Math.abs(curr.value));
     curr = scanner.next();
     break;

case 70: // frozen layer
     layer.visible = (curr.value != 1 &&  curr.value != 2 );
     curr = scanner.next();
     break;
```

**Comment by kpetrow** [comment]
@bzuillsmith I can do a PR for this too.

**Comment by bzuillsmith** [comment]
A PR would be perfect, preferably with a test. Modifying one of the existing tests would be fine. Thanks!

**Comment by bzuillsmith** [comment]
I went ahead and did that work for this. I will probably publish it this weekend.

**Comment by bzuillsmith** [comment]
Published in 0.5.1
